components/TodoForm.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useRef, useContext } from "react";
+import TodoModel from "../models/todo";
+import { TodoContext } from "../store/store-todo";
+import classes from "../styles/Home.module.css";
+
+const TodoForm = () => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const todoCtx = useContext(TodoContext);
+  const addTodo = todoCtx.addTodo;
+
+  const submitHandler = (e: React.FormEvent) => {
```

Comment:
1. Try splitting the validate and call API parts out of each other.
2. add test the validate part too